src/mqt/qecc/co3/microscopic/snake_builder.py

```diff
@@ -0,0 +1,1389 @@
+                for label in weight_four["i+1"]:
+                    for key, val in next_patch.items():
+                        if label == val:
+                            stab_temp.update({key: val})
+            z_stabilizers.append(stab_temp)
+
+        self.z_stabilizers = z_stabilizers
+        return z_stabilizers
+
+    @staticmethod
+    def find_matching_dict_z(
+        x_stabs: list[dict[tuple[int, int], int]],
+        current_patch: dict[tuple[int, int], int],
+        next_patch: dict[tuple[int, int], int],
+    ) -> dict[tuple[int, int], int] | None:
+        """Finds the x stabilizer which connects current_patch and next_patch."""
+        current_keys = set(current_patch.keys())
+        next_keys = set(next_patch.keys())
+
+        for x_stab in x_stabs:
+            x_keys = set(x_stab.keys())
+            if x_keys & current_keys and x_keys & next_keys:  # Check intersection with both
+                return x_stab
+        return None  # Return None if no match is found
+
+    def find_disjoint_dict(
+        self,
+        current_patch: dict[tuple[int, int], int],
+        next_patch: dict[tuple[int, int], int],
+        dict_list: list[dict[str, list[int]]],
+        pair: list[int],
+    ) -> dict[str, list[int]] | None:
+        """Finds a dictionary in which the total of integers in the values is disjoint to the given pair. AND the weight4 must bepairwise neighbors."""
+        pair_set = set(pair)  # Convert the pair to a set for easy comparison
+
+        def neighboring_pair(
+            current_patch: dict[tuple[int, int], int],
+            next_patch: dict[tuple[int, int], int],
+            dictionary: dict[str, list[int]],
+        ) -> bool | None:
+            """Checks whether at least one neighboring pair in chosen 4-weight stab."""
+            for label_i in dictionary["i"]:
+                for label_i1 in dictionary["i+1"]:
+                    # Find positions in the dictionaries
+                    positions_i = [pos for pos, label in current_patch.items() if label == label_i]
+                    positions_i1 = [pos for pos, label in next_patch.items() if label == label_i1]
+
+                    # Check if any position from i is a neighbor of a position from i+1
+                    for pos_i in positions_i:
+                        for pos_i1 in positions_i1:
+                            if self.g.has_edge(pos_i, pos_i1):  # Check for an edge
+                                return True
+            return False
+
+        for dictionary in dict_list:
+            # Get all integers in the dictionary's values
+            value_set = {val for values in dictionary.values() for val in values}
+            # Check if the two sets are disjoint
+            if pair_set.isdisjoint(value_set) and neighboring_pair(current_patch, next_patch, dictionary):
+                return dictionary
+
+        return None
+
+    def translate_checks(self) -> tuple[list[list[int]], list[list[int]]]:
```

Comment:
There is some inconsistency here. We have `gen_checks`, `gen_check_matrix` and `translate_checks` in the different classes, respectively. 

Also is there a reason these are lists instead of numpy arrays?